iree/test/e2e/regression/CMakeLists.txt

```diff
@@ -248,4 +248,96 @@ iree_generated_trace_runner_test(
     "aarch64:+dotprod"
 )
 
+iree_generated_trace_runner_test(
```

Comment:
Based on recently found information, these tests have a problem. They are being added with a focus on ARM codegen, but on CI they are run on x86. The x86 path and the ARM path are diverging at this point (maybe eventually they will converge again, but different groups, different approaches). These tests are now inter-mixing them. The fallout will be
1) Any change to x86 codegen will affect these tests, adding burden on the x86 path. 
2) Any test you are adding here is taking a different compilation path on CI than what you are developing on, so any development work on this path will also have to account for what happens on x86 path adding burden to you guys.

In the end they are all tests, and they all need to pass, but tests also reflect status of work, and these tests are not capturing the latter.

I'd suggest making these tests running only on android (or at least run on android in addition to x86). I am actually less concerned about these tests using x86 compilation on CI. There is incentive to make this work for x86 anyways. More concerned that what you are adding these tests for during development is not what the CI is testing.

Reply:
Well on different CI they are run on different targets. The android CI is postsubmit-only, and that's going to remain the case till we set up qemu tests, but there's not really any way around that. We do local testing on android on these and the additional x86 presubmit can catch obvious structural things

Reply:
there's also the question of how narrow a supported path you really want to build here. I thought that we wanted to replace the old NN-inference-engine-as-a-handwritten-library, and its abrupt support cliffs, with a compiler approach that would offer more consistent support for what the community throws at it going forward. Is that still the plan?

mmt4d-based MLIR may not be exactly the flavor of MLIR that everyone else is primarily interested in testing, but mmt4d is just a (named) linalg generic. if the simple transformation that is matmul-to-mmt4d is enough to fall off of the supported path, isn't that of concern?

Reply:
There are different levels of concerns. The x86 path is changing radically with the move to sandbox. It will come with its own issues. I am not saying that these tests shouldn't test that path. Happy that they did, and found issues. It is strange to me though that these tests are added for one purpose of development but the CI is testing it on generic x86 compilation which is the codegen driver (from sandbox). That path triggers vectorization in different ways that you are probably not familiar with / have not looked into. Yet it is tested on CI. This means things are strangely intertwined in a way that no body is looking at /owns the problems.

For now we have carved out a different compilation path for mmt4d so that it doesnt go down codegen driver. I really wish we didnt have to do that.

Reply:
i never understood what you mean with
> have not looked into. Yet it is tested on CI. This means things are strangely intertwined in a way that no body is looking at /owns the problems

in my book, what is current on CI is what every new code change must pass ; so any CI failure caused by a new PR is by definition looked at by the authors of the new PR, which, in this case, would be you :-) We can argue about how good a use of your time that is, but I don't see the "nobody is looking at / owns the problems" part.

Reply:
Noting again that the CI is testing this on both arm and x86. It's just that the arm build doesn't run on presubmit because of resource limitations.

Reply:
FTR : The PR that switched the x86 codegen path to use sandbox codegen driver passed the CI. Its not that a change was made without having tests passing. Why these tests didnt fail when compilation triggered an infinite loop is a mystery to me.

> so any CI failure caused by a new PR is by definition looked at by the authors of the new PR, which, in this case, would be you :-) We can argue about how good a use of your time that is, but I don't see the "nobody is looking at / owns the problems" part.

You are missing the point here. The mmt4d path is doing vectorization in a way that is different from the x86 codegen path that uses the codegen driver. The latter is very deliberate in how it triggers various vectorization patterns, that might be different from where you expect the MMT4d vectorization patterns to trigger. You are looking at only ARM codegeneration during development, but CI is testing x86. Which means you (might) need to triage issues that occur in CI but not during your development due to the different vectorization paths. Consequently, I dont see a "real" value in testing mmt4d path on x86. Its good for general correctness testing (so stating again for the nth time if it wasnt clear, I am fine with these tests running in CI, more saying this is potentially undue burden on people working with mmt4d), but no one is working on mmt4d path for x86 code-generation.

Reply:
Re the infinite loop -- I see, I too have been running into infinite loops around MLIR core patters around Vector dialect lowerings (I think --- it has been hard to pin down exactly).  My experience generally has been that in order to use Vector dialect without running into such hard-to-debug issues, it is best to use it exactly like its owners in MLIR core are using it themselves. So if I'm going to take away something high-level from this discussion, it's going to be a confirmation of that.  And so, now, I too would like to be on the codegen driver.  If mmt4d were on the codegen driver, would that resolve your concern here?

I am trying to keep mmt4d it itself generic, not ARM-specific. It was news to me in this conversation that anyone would treat it as ARM-specific. I understand that matmul efforts on the MLIR core side have not been interested in mmt4d, and that's fine --- but I thought, we thought that matmul was this complex, multidisciplicary, critical topic where we are intentionally investing in more than one approach for the time being. We were always going to eventually arbitrate on this, but I can think of a bunch of milestones that would need to be reached first before one could do so meaningfully (e.g. implement i8 matmul on x86,  which will force 2d tiling topics into the discussion by virtue of the shape of the pmaddwd x86 instruction).

Reply:
Maybe it would be good for all of us to meet to discuss this in a higher-bandwidth setting